src/weights.jl

```diff
@@ -210,9 +210,12 @@ end
     eweights(n::Integer, λ::Real; scale=false)
 
 Construct a [`Weights`](@ref) vector which assigns exponentially decreasing weights to past
-observations, which in this case corresponds to larger integer values `i` in `t`.
-If an integer `n` is provided and `t` is omitted, weights are generated for values from 1 to
-`n` (equivalent to `t = 1:n`).
+observations (larger integer values `i` in `t`).
+The integer value of `n` represents the number of past observations to consider, which defaults to:
+
+1. `maximum(t) - minimum(t) + 1` if only `t` is passed in and the elements are integers
+2. `length(r)` if a superset range `r` is passed in
+3. `n` is explicitly passed instead of `t` (equivalent to `t = 1:n`)
```

Comment:
This describes default values for both `n` and `t`, so I suggest changing the wording a bit. For example:
```suggestion
The integer value `n` represents the number of past observations to consider.
`n` defaults to `maximum(t) - minimum(t) + 1` if only `t` is passed in
and the elements are integers, and to `length(r)` if a superset range `r`
is also passed in.
If `n` is explicitly passed instead of `t`, `t` defaults to `1:n`.
```